src/tribler/core/components/popularity/popularity_component.py

```diff
@@ -22,13 +24,16 @@ async def run(self):
         metadata_store_component = await self.require_component(MetadataStoreComponent)
         torrent_checker_component = await self.require_component(TorrentCheckerComponent)
 
+        rendezvous_db = RendezvousDatabase(db_path=self.session.config.state_dir / STATEDIR_DB_DIR / PopularityCommunity.RENDEZVOUS_DB_NAME)
```

Comment:
Does this require a separate `Component` instead?

Reply:
No, it's perfectly acceptable to create a database here. However, don't forget to close the database if it's necessary.

https://github.com/Tribler/tribler/blob/3b760abf3f5d5d29a84d8d0c27df543f07fc7084/src/tribler/core/components/knowledge/knowledge_component.py#L57-L58